NIX.md

```diff
@@ -0,0 +1,138 @@
+# BDK's Nix Guidelines
+
+This document outlines:
+
+1. What is Nix and how to install.
+1. BDK's Nix-based tests and CI.
+1. Developer best-practices with Nix.
+1. How to run replicate CI tests locally.
+
+## Nix
+
+We use [Nix](https://nixos.org/) as the CI tool for BDK and also as a
+development environment for the project.
+Nix is purely functional.
+Everything is described as an expression/function,
+taking some inputs and producing deterministic outputs.
+This guarantees reproducible results and makes caching everything easy.
+Nix expressions are lazy. Anything described in Nix code will only be executed
+if some other expression needs its results.
+This is very powerful but somewhat unnatural for developers not familiar
+with functional programming.
+
+There are several resources to get started with Nix:
+
+- [NixOS Wiki](https://nixos.wiki/).
+- [Nix Reference Manual](https://nixos.org/manual/nix/stable/).
+- [`nixpkgs` Manual](https://nixos.org/manual/nixpkgs/stable/).
+- [Official documentation for getting things done with Nix](https://nix.dev/).
+
+There's also the [`nix-bitcoin` project](https://nixbitcoin.org/).
+
+To install Nix please follow the [official Nix documentation](https://nixos.org/manual/nix/stable/installation/installation.html).
+You may need to enable Flakes support, check the instructions at [NixOS Wiki entry on Flakes](https://nixos.wiki/wiki/Flakes).
```

Comment:
@storopoli wdyt about also mentioning the https://github.com/DeterminateSystems/nix-installer ?
it can be a helpful alternative way to install/remove it :)

Reply:
Yes, despite being my preferred way, it is not the official installer.
Maybe a footnote what you think?

Reply:
Sure, a footnote is good enough

Reply:
https://github.com/cachix/devenv/issues/509#issuecomment-1919469287 @storopoli they are considering it